File: frontend/src/api/jobs.ts
import { apiGet } from "./client";

export type JobStatus = {
  job_id: string;
  job_type: string;
  status: string;
  payload: Record<string, unknown>;
  result: Record<string, unknown> | null;
  error_message: string | null;
  created_at: string;
  updated_at: string;
};

export function fetchJobStatus(jobId: string) {
  return apiGet<JobStatus>(`/api/v1/jobs/${jobId}`);
}

/** Latest queued/running maps_scan for the signed-in client (404 when idle). */
export function fetchActiveScanJob() {
  return apiGet<JobStatus>("/api/v1/jobs/active/scan");
}

File: backend/app/schemas/jobs.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class JobStatusResponse(BaseModel):
    job_id: UUID
    job_type: str
    status: str
    payload: dict
    result: dict | None
    error_message: str | None
    created_at: datetime
    updated_at: datetime


class ScanCreateRequest(BaseModel):
    keyword: str | None = None
    radius_km: int | None = None
